components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -271,6 +277,26 @@ public boolean handleRequest(MessageContext messageContext) {
         }
     }
 
+    public static TokenGenerator getTokenGenerator() {
+        if (tokenGenerator == null) {
+            AppManagerConfiguration configuration= org.wso2.carbon.appmgt.impl.service.ServiceReferenceHolder.getInstance()
+                    .getAPIManagerConfigurationService().getAPIManagerConfiguration();
+            if (configuration == null) {
```

Comment:
Log to error is OK for now. we need to change getAPIManagerConfiguration() to throw an appropriate exception when returns null due to configuration file error. (Separate task)